tools/analyzer_plugin/test/integration/assists/toggle_statefulness_test.dart

```diff
@@ -0,0 +1,151 @@
+import 'dart:async';
+
+import 'package:analyzer_plugin/utilities/assist/assist.dart';
+import 'package:over_react_analyzer_plugin/src/assist/toggle_stateful.dart';
+import 'package:test/test.dart';
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../test_bases/assist_test_base.dart';
+import 'boilerplate_assist_utils.dart';
+
+void main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(AddStatefulnessAssist);
+    defineReflectiveTests(RemoveStatefulnessAssist);
+  });
+}
+
+@reflectiveTest
+class AddStatefulnessAssist extends AssistTestBase with BoilerplateAssistTestStrings {
+  static int expectedPriority = ToggleComponentStatefulness.statefulAssistPriority;
+
+  @override
+  AssistKind get assistKindUnderTest => ToggleComponentStatefulness.makeStateful;
+
+  Future<void> test_noAssistOnFactory() async {
+    final generatedSource = simpleUiComponentSource();
+
+    var source = newSource(BoilerplateAssistTestStrings.fileName, generatedSource);
+    var selection = createSelection(source,
+        generatedSource.replaceFirst(r'UiFactory<FooProps> Foo = _$Foo;', r'UiFactory<FooProps> #Foo# = _$Foo;'));
```

Comment:
Why the replacement and not just the following?
```suggestion
    var selection = createSelection(source, r'UiFactory<FooProps> #Foo# = _$Foo;');
```
The whole source isn't needed to make a selection; the context before/after can be as small/large as you want